src/odemis/driver/technolution.py

```diff
@@ -518,45 +438,136 @@ def __init__(self, name, role, parent, **kwargs):
+        y_setpoints = scan_offset_volts[1] + scan_amplitude_volts[1] * signal.sawtooth(
+                2 * math.pi * calibration_frequency * time_points_scanner)
+
+        return x_setpoints.tolist(),\
+               y_setpoints.tolist(),\
+               tick_calibration_dwell_time
+
+    def _calc_calibration_sampling_period(self, total_line_scan_time, system_clock_period, descanner_clock_period):
+        """
+        For creation of the calibration setpoints the sampling period is variable but the total number of
+        setpoints is limited to maximum 9000 points. Depending on the total line scanning time the calibration
+        setpoints have a variable sampling period (calibration dwell time) which is a multiple of the system
+        clock period. The number of setpoints (and the corresponding sampling period is chosen such that the
+        biggest number of setpoints is used (highest resolution of the signal).
+        The sampling period is defined by finding the higest number of setpoints for the calibration signal
+        with a sampling period equal to a whole multiple of the system clock period. This is done by iteratively
+        increasing the sampling period.
+
+        :param total_line_scan_time (float): Total line scanning time in seconds
+        :param system_clock_period (float): clock period of the system and scanner in seconds
+        :param descanner_clock_period (float): Descanner lock period of the scanner in seconds
+        :return: (int) Sampling period in ticks
+                 (float) Sampling period is seconds (not used yet)
+                 (int) Number of setpoints
+        """
+        # TODO MAX_NMBR_POINT value of 9000 is not sufficient for the entire range of the dwell time, this value
+        #  needs to be increased on the ASM HW.
+        MAX_NMBR_POINT = 9000  # Constant maximum number of setpoints
+
+        # Calculate the range (minimum and maximum multiplication factor) in which to search for the best
+        # calibration dwell time/sampling period.
+
+        # The minimal number of ticks of the system clock period to get a sampling period with max 9000 setpoints
+        min_sampling_period_ticks = int(numpy.ceil((total_line_scan_time / MAX_NMBR_POINT) / system_clock_period))
+        # If the found minimum sampling period (in ticks) is smaller than the minimal dwell time for the scanner,
+        # use the minimum dwell time in ticks.
+        min_sampling_period_ticks = max(min_sampling_period_ticks, int(self.dwellTime.range[0] / system_clock_period))
+
+        # The max sampling period in ticks is defined by the descanner clock period
+        max_sampling_period_ticks = int(descanner_clock_period / system_clock_period)
+
+        for sampling_period_ticks in range(min_sampling_period_ticks, max_sampling_period_ticks):
+            sampling_period = sampling_period_ticks * system_clock_period  # Increase the sampling period
```

Comment:
This comment does not make sense. You simply calculate the sampling period in seconds.

Reply:
Just remove it